pkg/platform/yaml/yaml.go

```diff
@@ -129,57 +130,81 @@ func (ys *Yaml) Write(path, fileName string, docRead platform.KindSpecifier) err
 	return nil
 }
 
-func containsMatchingUrl(urlMethods map[string][]string, urlStr string, requestMethod models.Method) bool {
-	parsedURL, err := url.Parse(urlStr)
+func containsMatchingUrl(urlMethods []string, urlStr string, requestUrl string, requestMethod models.Method) (error, bool) {
+	urlMatched := false
+	parsedURL, err := url.Parse(requestUrl)
 	if err != nil {
-		return false
+		return err, false
 	}
 
 	// Check for URL path and method
-	path := parsedURL.Path
-	if methods, exists := urlMethods[path]; exists {
-		// Loop through the methods for this path
-		for _, method := range methods {
-			// If the request method matches one of the allowed methods, return true
+	regex, err := regexp.Compile(urlStr)
+	if err != nil {
+		return err, false
+	}
+
+	urlMatch := regex.MatchString(parsedURL.Path)
+
+	if urlMatch && len(urlStr) != 0 {
+		urlMatched = true
+	}
+
+	if len(urlMethods) != 0 && urlMatched {
+		urlMatched = false
+		for _, method := range urlMethods {
 			if string(method) == string(requestMethod) {
-				return true
+				urlMatched = true
 			}
 		}
-		// If the request method is not in the allowed methods, return false
-		return false
 	}
 
-	return false
+	return nil, urlMatched
 }
 
-func hasBannedHeaders(object map[string]string, bannedHeaders []string) bool {
-	for headerName, _ := range object {
-		for _, bannedHeader := range bannedHeaders {
-			if headerName == bannedHeader {
-				return true
+func hasBannedHeaders(object map[string]string, bannedHeaders map[string]string) (error, bool) {
+	for headerName, headerNameValue := range object {
+		for bannedHeaderName, bannedHeaderValue := range bannedHeaders {
+			regex, err := regexp.Compile(headerName)
+			if err != nil {
+				return err, false
+			}
+			headerNameMatch := regex.MatchString(bannedHeaderName)
+
+			regex, err = regexp.Compile(bannedHeaderValue)
+			if err != nil {
+				return err, false
+			}
+			headerValueMatch := regex.MatchString(headerNameValue)
+			if headerNameMatch || headerValueMatch {
```

Comment:
Shouldn't this be a and condition..?

Reply:
earlier I thought of using either header name or value to be independent, will update it so that the combination is unique(i.e, the and condition)